setup.py

```diff
@@ -21,7 +21,12 @@
 COMPILER_DIRECTIVES = {}
 # This Cython macro disables a build warning, obsolete with Cython>=3
 #  see : https://cython.readthedocs.io/en/latest/src/userguide/migrating_to_cy30.html#numpy-c-api
-DEFINE_MACROS = [("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")]
+DEFINE_MACROS = [
+    ("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION"),
+    # 0x030B0000 -> 3.11
+    ("Py_LIMITED_API", "0x030B0000"),
```

Comment:
Assuming 3.11 is liable to move forward, there should be a line in `test_python_versions()`:

https://github.com/SciTools/cf-units/blob/9e34057547f407df84fc3d5bc742f93831cbbe15/cf_units/tests/test_coding_standards.py#L83-L159